src/main/java/me/jellysquid/mods/lithium/mixin/entity/fast_piglin_brain/PiglinBrainMixin.java

```diff
@@ -0,0 +1,89 @@
+package me.jellysquid.mods.lithium.mixin.entity.fast_piglin_brain;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.At;
+
+import net.minecraft.entity.mob.PiglinBrain;
+import net.minecraft.entity.mob.PiglinEntity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.ai.brain.Activity;
+import net.minecraft.entity.ai.brain.task.LookTargetUtil;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.world.GameRules;
+
+@Mixin(PiglinBrain.class)
+public class PiglinBrainMixin {
+
+    /**
+     * The vanilla implimentation here seems to be heavy usage of
+     * functional style java and similar which performs poorly when
+     * used in hot code
+     * 
+     * This optimization here helps to reduce a spike in lag from 
+     * breaking a specific block when near a bunch of piglins
+     * 
+     * @param player the player that broke the block
+     * @param isInventory if the player broke a block that stores items 
+     */
+    @Inject(method = "onGuardedBlockBroken", at = @At("HEAD"), cancellable = true)
+    public static void onGuardedBlockBroken(PlayerEntity player, boolean isInventory, CallbackInfo ci) {
```

Comment:
The game complains that static methods must be private
org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: Mixin lithium.mixins.json:entity.fast_piglin_brain.PiglinBrainMixin contains non-private static method onGuardedBlockBroken